openlibrary/catalog/add_book/__init__.py

```diff
@@ -678,10 +677,369 @@ def load_data(rec, account_key=None):
     return reply
 
 
-def load(rec, account_key=None):
-    """Given a record, tries to add/match that edition in the system.
+def add_missing_int_and_str_values_to_edition(
+    source_rec: dict,
+    edition: Edition,
+    need_edition_save: bool,
```

Comment:
Can we make this not a parameter? It might be clearer if all of these functions just returned a bool for `need_edition_save` that just basically says if the current function made changes to the edition. Then the parent function can `or` all those together to determine if it needs to save.